### PR TITLE
feature(ruff): check for `unused-import (F401)`

### PR DESCRIPTION
**Pull request by fruch**
we had this check in pylint, and since we missing it on ruff it casue linting errors in backport PRs

this is something we want to keep checked anyhow

Ref: https://docs.astral.sh/ruff/rules/unused-import/

### Testing

- [x] provision tests
- [x] integration tests

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: sdcm/cluster_baremetal.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def __init__(self, **kwargs):
         cluster.BaseLoaderSet.__init__(self, kwargs["params"])
         PhysicalMachineCluster.__init__(self, **kwargs)
 
-    @ classmethod
```
Comment (fruch) [inline sdcm/cluster_baremetal.py:169]: I had updated version of autopep8 that spotted this...

**Comment by fruch** [comment]
Some integration tests failing
```
01:35:28 FAILED unit_tests/test_cql_stress_cassandra_stress_thread.py::test_01_cql_stress_cassandra_stress - IndexError: list index out of range 01:35:28 FAILED unit_tests/test_cql_stress_cassandra_stress_thread.py::test_02_cql_stress_cassandra_stress_multi_region - IndexError: list index out of range
```

Integration tests should not make block provision tests !!! 😭

**Review by soyacz — APPROVED** [review]
LGTM
Why do you think it requires provision and integration tests?

**Comment by fruch** [comment]
> LGTM
> Why do you think it requires provision and integration tests?

Cause of imports done from those __init__.py

When working last time on ruff I was bitten by that a few times

**Pull request merged**